1.19/Dockerfile

```diff
@@ -11,24 +11,30 @@ LABEL maintainer="Community & Partner Engineering Team <community-partner@circle
 
 ENV GO_VER="1.19.10"
 
+USER root
+
 # Install packages needed for CGO
-RUN sudo apt-get update && sudo apt-get install -y --no-install-recommends \
+RUN apt-get update && apt-get install -y --no-install-recommends \
 		g++ \
 		libc6-dev && \
-	sudo rm -rf /var/lib/apt/lists/* && \
+	rm -rf /var/lib/apt/lists/* && \
 	# Check if we're facing the first minor release issue and adjust if needed
 	if [[ ${GO_VER:(-2)} == ".0" ]]; then \
 	  GO_VER=${GO_VER:0:-2}; \
 	fi && \
-	curl -sSL "https://go.dev/dl/go${GO_VER}.linux-amd64.tar.gz" | sudo tar -xz -C /usr/local/ && \
+	[[ $(uname -m) == "x86_64" ]] && ARCH="amd64" || ARCH="arm64" && \
+	curl -sSL "https://go.dev/dl/go${GO_VER}.linux-${ARCH}.tar.gz" | tar -xz -C /usr/local/ && \
 	mkdir -p /home/circleci/go/bin
```

Comment:
the dockerfile.template needs to add `chown -R 1001:1002 /home/circleci` otherwise anyone trying to build a go project won't be able to write a cache to the gopath

https://app.circleci.com/pipelines/github/CircleCI-Public/cimg-go/689/workflows/d6f6ca8a-e72a-4fd1-a4ed-95e933d0be40

https://app.circleci.com/pipelines/github/CircleCI-Public/sample-go-cfd/32/workflows/bdbdfd67-2e97-4841-839f-f11a2df88d7c

Reply:
Good catch. For now we will revert and release as normal, I feel other images are going to have worse problems with permissions so we probably need to re-evaluate how the images are being built before proceeding